sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -33,18 +34,30 @@ protected LogsIngestionClient()
         // If no concurrency count is provided for a parallel upload, default to 5 workers.
         private const int DefaultParallelWorkerCount = 5;
 
-        internal readonly struct BatchedLogs <T>
+        internal readonly struct BatchedLogs
         {
-            public BatchedLogs(int logsCount, BinaryData logsData)
+            public BatchedLogs(List<object> logs, BinaryData logsData)
             {
-                LogsCount = logsCount;
+                Logs = logs;
                 LogsData = logsData;
             }
 
-            public int LogsCount { get; }
+            public List<object> Logs { get; }
             public BinaryData LogsData { get; }
         }
 
+        internal readonly struct ProcessEventHandler
```

Comment:
Isn't ShouldAbort based on whether the user throws an exception from their handler? If that's the case, all we need to return is the exception from our method that processes the completed task, rather than returning this type.